### PR TITLE
[FIX] test_mail: fix activity test failing randomly

### PR DESCRIPTION
**Pull request by pyduf**
Each failing of the test "activity menu widget: activity menu with 2 models"
was close to midnight. And that was actually the cause of the error.

Actually, when executing that test very close to midnight, the test activities
are set up for a day and the check happens the day after (actually a few second
after). As we check in this test "overdue", "planned" and "today" activities,
the tested value are not inline anymore with the test setup when switching from
a day to the next (passing midnight). For example, "today" activities become
"overdue" activities. To avoid that we patch the date to be far from midnight.

[IMP] test_mail: use "await contains" instead of "assert" on dom

We replace the immediate dom checks with the more rebust and standard construct
"await contains".

Task-3592798

### CODE REVIEW

File: addons/test_mail/static/tests/systray_activity_menu_tests.js [hunk 1]
```diff
@@ -94,24 +88,22 @@ QUnit.test("activity menu widget: activity menu with 2 models", async (assert) =
...
-        containsN(document, ".o-mail-ActivityMenu .o-mail-ActivityGroup", ["mail.test.activity", "1 Today"])
-    );
-    assert.ok(
-        containsN(document, ".o-mail-ActivityMenu .o-mail-ActivityGroup", ["mail.test.activity", "2 Future"])
-    );
+    await contains(".o-mail-ActivityMenu .o-mail-ActivityGroup", {
+        contains: [
+            ["div[name='activityTitle']", { text: "res.partner" }],
+            ["span", { text: "0 Late" }],
+            ["span", { text: "1 Today" }],
```
Comment (awa-odoo) [inline addons/test_mail/static/tests/systray_activity_menu_tests.js:100]: I'm curious, does this specifically check that we have "1 Today" **for res.partner** ?

As it seems to independently check each selector, so you could have "1 Today" for "mail.test.activity" and this check would still be green, right?

I was probably already the case before though, maybe not needed to change everything.

**Review by awa-odoo — COMMENTED** [review]
Hello @pyduf 

Globally LGTM, left one typo comment and one question.

Thanks for handling this one so quickly and cheers!

File: addons/test_mail/static/tests/systray_activity_menu_tests.js (hunk 1 above)
Comment (seb-odoo) [reply addons/test_mail/static/tests/systray_activity_menu_tests.js:100]: Hello, just answering as I'm passing-by, all the sub-contains need to be all children of the same `.o-mail-ActivityMenu .o-mail-ActivityGroup` at the same time.
So it specifically checks for one notification item that has res.partner, 0 late and 1 today at the same time.

The custom implementation of `containsN` that was in this test before the diff made the same kind of check actually.

**Comment by pyduf** [comment]
> Hello @pyduf
> 
> Globally LGTM, left one typo comment and one question.
> 
> Thanks for handling this one so quickly and cheers!

Hello @awa-odoo , 

Thanks for the quick review. I have corrected the typo in the commit message.

File: addons/test_mail/static/tests/systray_activity_menu_tests.js (hunk 1 above)
Comment (awa-odoo) [reply addons/test_mail/static/tests/systray_activity_menu_tests.js:100]: Hello @seb-odoo 

Ok great then, thanks for the explanations!

**Comment by awa-odoo** [comment]
@robodoo r+ rebase-ff

**Comment by tde-banana-odoo** [comment]
@d-fence fix for the failing activity test around midnight :)

**Pull request closed without merging**